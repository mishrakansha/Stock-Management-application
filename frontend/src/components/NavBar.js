import React, { Component } from "react";
import TextField from "@mui/material/TextField";
import "./css/NavBarCss.css";
import SideNavBar from "./SideNavBar";
import NewItemForm from "./forms/NewItemForm";
import DashboardPage from "./DashboardPage";

import { Route, Routes } from "react-router-dom";

export class NavBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hamburgerOpen: false,
    };
    // const { window } = this.props;
  }
  hamburgerClick = async () => {
    console.log("clicked");
    this.setState({
      hamburgerOpen: !this.state.hamburgerOpen,
    });
  };
  componentDidUpdate() {
    console.log(document.querySelector("#dataContainer"));
    this.state.hamburgerOpen
      ? (document.querySelector("#dataContainer").style = "grid-column: 2/7;")
      : (document.querySelector("#dataContainer").style = "grid-column: 1/7;");
  }
  render() {
    return (
      <>
        <div className="headnavbar">
          <nav className="nav">
            <ul className="leftsidelist">
              <li onClick={this.hamburgerClick} className="baricon">
                <i className="fa-solid fa-bars"></i>
              </li>
              <li>
                <div className="icon">Stock management app</div>
              </li>
            </ul>
            <ul className="searchbar">
              <span className="searchicon">
                <i className="fas fa-search "></i>
              </span>
              <TextField
                variant="outlined"
                hiddenLabel
                id="filled-hidden-label-small"
                placeholder="search item.........."
                size="small"
                inputProps={{
                  style: {
                    border: "none",
                    padding: "8.5px 14px 8.5px 30px",
                  },
                }}
              />{" "}
            </ul>
          </nav>
        </div>

        {this.state.hamburgerOpen && <SideNavBar />}
        <div id="dataContainer">
          <Routes>
            <Route
              exact
              path="/AddItem"
              element={<NewItemForm key="addItem" />}
            ></Route>
            <Route
              exact
              path="/"
              element={<DashboardPage key="DashBoard" />}
            ></Route>
          </Routes>
        </div>
      </>
    );
  }
}

export default NavBar;
