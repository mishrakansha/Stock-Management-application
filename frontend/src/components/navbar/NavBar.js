import React, { Component } from "react";
import "./navBar.css";
import SideNavBar from "../sidebarnav/SideNavBar";
import { ExpendableSlidebar } from "../sidebarnav/SideNavBar.js";
import { Outlet } from "react-router-dom";
import PropTypes from "prop-types";
export class NavBar extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showhamburger: window.innerWidth > 760,
    };
  }
  hamburgerClick = async () => {
    const { isExpended, dataContainer, sideNavBarExpended } = this.props;
    if (window.innerWidth > 760) {
      if (!isExpended) {
        dataContainer({ gridColumn: "3/7" });
        sideNavBarExpended(true);
      } else {
        dataContainer({ gridColumn: "2/7" });
        sideNavBarExpended(false);
      }
    }
  };
  componentDidMount() {
    const { dataContainer, sideNavBarExpended } = this.props;
    window.addEventListener("resize", (event) => {
      const width =
        document.documentElement.clientWidth ||
        document.body.clientWidth ||
        window.innerWidth;
      if (width > 760) {
        this.setState({ showhamburger: true });
      } else {
        dataContainer({ gridColumn: "2/7" });
        sideNavBarExpended(false);
        this.setState({ showhamburger: false });
      }
    });
  }

  render() {
    const { isExpended } = this.props;
    const { showhamburger } = this.state;
    return (
      <>
        <div className="headNavBar">
          <nav className="nav">
            <ul className="leftSideList">
              {showhamburger && (
                <li
                  onClick={this.hamburgerClick}
                  id="hamburger"
                  className="barIcon"
                >
                  <i className="fa-solid fa-bars"></i>
                </li>
              )}
              <li>
                <div className="navHeading">Stock management app</div>
              </li>
            </ul>

            <ul className="profileIcon">
              <li className="barIcon">
                <i className="fa-solid fa-user"></i>
              </li>
            </ul>
          </nav>
        </div>
        {isExpended && window.innerWidth > 760 ? (
          <ExpendableSlidebar />
        ) : (
          <SideNavBar />
        )}{" "}
        <Outlet />
      </>
    );
  }
}
NavBar.propTypes = {
  dataContainer: PropTypes.func,
  sideNavBarExpended: PropTypes.func,
  isExpended: PropTypes.bool,
  dataContainerGrid: PropTypes.object,
};
export default NavBar;
