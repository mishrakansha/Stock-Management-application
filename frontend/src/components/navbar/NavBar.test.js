import React from "react";
import { shallow } from "enzyme";
import NavBar from "./NavBar";
describe("NavBar", () => {
  it("Take a snapshot if do not added", () => {
    const component = shallow(<NavBar />);
    expect(component).toMatchSnapshot();
  });
  it("Test to render Navbar Heading", () => {
    const component = shallow(<NavBar />);
    expect(component.find(".navHeading").text()).toEqual(
      "Stock management app"
    );
  });
});
