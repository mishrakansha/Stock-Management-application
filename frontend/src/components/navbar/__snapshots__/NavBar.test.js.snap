// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Take a snapshot if do not added 1`] = `
<Fragment>
  <div
    className="headNavBar"
  >
    <nav
      className="nav"
    >
      <ul
        className="leftSideList"
      >
        <li
          className="barIcon"
          id="hamburger"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-bars"
          />
        </li>
        <li>
          <div
            className="navHeading"
          >
            Stock management app
          </div>
        </li>
      </ul>
      <ul
        className="profileIcon"
      >
        <li
          className="barIcon"
        >
          <i
            className="fa-solid fa-user"
          />
        </li>
      </ul>
    </nav>
  </div>
  <SideNavBar />
   
  <Outlet />
</Fragment>
`;
